cypress-tests/cypress/e2e/privateProject.cy.ts

```diff
@@ -0,0 +1,157 @@
+
+const projectIdentifier: ProjectIdentifier = {
+  name: projectTestConfig.projectName,
+  namespace: username,
+};
+
+describe("Basic public project functionality", () => {
+  before(() => {
+    // Use a session to preserve login data
+    cy.session(
+      "login-publicProject",
+      () => {
+        cy.robustLogin();
+      },
+      validateLogin
+    );
+
+    // Create a project for the local spec
+    if (projectTestConfig.shouldCreateProject) {
+      cy.visit("/");
+      cy.createProject({ templateName: "Python", ...projectIdentifier, visibility: "private" });
+    }
+  });
+
+  beforeEach(() => {
+    // Restore the session
+    cy.session(
+      "login-publicProject",
+      () => {
+        cy.robustLogin();
+      },
+      validateLogin
+    );
+    cy.visitAndLoadProject(projectIdentifier);
+  });
+
+  it("Can search for project only when logged in", () => {
+    // Assess the project has been indexed properly
+    cy.dataCy("project-navbar", true)
+      .contains("a.nav-link", "Settings")
+      .should("be.visible")
+      .click();
+    cy.dataCy("project-settings-knowledge-graph")
+      .contains("Project indexing", { timeout: TIMEOUTS.vlong })
+      .should("exist");
+    cy.dataCy("kg-status-section-open").should("exist").click();
+    cy.dataCy("project-settings-knowledge-graph")
+      .contains("Everything indexed", { timeout: TIMEOUTS.vlong })
+      .should("exist");
+    cy.dataCy("visibility-private").should("be.visible").should("be.checked");
+    cy.searchForProject(projectIdentifier, true);
+
+    // logout and search for the project and log back in
+    cy.logout();
+    cy.get("#nav-hamburger").should("be.visible").click();
+    cy.searchForProject(projectIdentifier, false);
+    cy.robustLogin();
+  });
+
+  it("Can always search for project after changing the visibility", () => {
+    // Change visibility to public
+    cy.dataCy("project-navbar", true)
+      .contains("a.nav-link", "Settings")
+      .should("be.visible")
+      .click();
+    cy.dataCy("project-settings-knowledge-graph")
+      .contains("Project indexing", { timeout: TIMEOUTS.vlong })
+      .should("exist");
+    cy.dataCy("visibility-private").should("be.visible").should("be.checked");
+    cy.dataCy("visibility-public").should("be.visible").check();
+    cy.get(".modal")
+      .contains("Change visibility to Public")
+      .should("be.visible");
+    cy.dataCy("update-visibility-btn").should("be.visible").click();
+    cy.get(".modal .alert-success", { timeout: TIMEOUTS.long })
+      .contains("The visibility of the project has been modified")
+      .should("be.visible");
+    // ? We need to wait before other checks take place.
+    // ? This is a workaround until we use the new Project update endpoint.
+    // ? Reference: https://github.com/SwissDataScienceCenter/renku-ui/issues/2778
+    cy.wait(10000); // eslint-disable-line cypress/no-unnecessary-waiting
```

Comment:
I would suggest using our existing constants for this.

```suggestion
    cy.wait(TIMEOUTS.standard); // eslint-disable-line cypress/no-unnecessary-waiting
```

Or, if it is important that we not wait longer than necessary, then 

`TIMEOUTS.standard * 0.5`